wpilib-config.cmake.in

```diff
@@ -15,3 +15,9 @@ find_dependency(Threads)
 @WPIMATH_DEP_REPLACE@
 @WPINET_DEP_REPLACE@
 @WPIUTIL_DEP_REPLACE@
+if(@WITH_JAVA@)
+    @WPILIBJ_DEP_REPLACE@
+    if(@WITH_WPIMATH@)
+        @WPIUNITS_DEP_REPLACE@
+    endif()
```

Comment:
If wpilibj is included, wpiunits must be as well (there's a dependency from wpilibj to wpiunits).

Reply:
No, it's supposed to be checking WITH_JAVA. You reminded me that I actually need to add this guard to wpimath as well. The reason for checking WITH_JAVA is because when building an external project, CMake will error out and won't even configure if WITH_JAVA isn't on, even in a C++ only project, because wpiunits doesn't get built, and wpimath's config won't be able to find wpiunits. The solution is to check if WITH_JAVA is on, and exclude wpiunits if it's off.

Reply:
Side note: for similar reasons, opting out of building any of the libraries (except sim plugins and gui) will cause `find_package(wpilib)` to fail. Is this intended? Or are users supposed to use find_package for the indiviual libraries in that case? If `find_package(wpilib)` should always work, should we add guards for each library that can be excluded?